### PR TITLE
enrich kubelet system oom event message info

### PR DESCRIPTION
**Pull request by gaorong**
**What this PR does / why we need it**:
enrich kubelet system oom event message info, add process name and pid.

before this PR:
```shell
[root@k8s-test-master01 ~]# kubectl describe node node1
Name:               node1
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=node1
                    kubernetes.io/os=linux
Annotations:        node.alpha.kubernetes.io/ttl: 0
....
Events:
  Type     Reason                   Age                   From            Message
  ----     ------                   ----                  ----            -------
  Normal   Starting                 17m                   kubelet, node1  Starting kubelet.
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered
  Normal   NodeAllocatableEnforced  17m                   kubelet, node1  Updated Node Allocatable limit across pods
  Normal   NodeHasSufficientMemory  17m                   kubelet, node1  Node node1 status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure    17m                   kubelet, node1  Node node1 status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID     17m                   kubelet, node1  Node node1 status is now: NodeHasSufficientPID
```

after this PR:
```shell
[root@k8s-test-master01 ~]# kubectl describe node node1
Name:               node1
Roles:              <none>
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    kubernetes.io/arch=amd64
                    kubernetes.io/hostname=node1
                    kubernetes.io/os=linux
Annotations:        node.alpha.kubernetes.io/ttl: 0
....
Events:
  Type     Reason                   Age                   From            Message
  ----     ------                   ----                  ----            -------
  Normal   Starting                 17m                   kubelet, node1  Starting kubelet.
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered, victim process: nginx, pid: 24804
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered, victim process: nginx, pid: 24840
  Warning  SystemOOM                17m                   kubelet, node1  System OOM encountered, victim process: nginx, pid: 128418
  Normal   NodeAllocatableEnforced  17m                   kubelet, node1  Updated Node Allocatable limit across pods
  Normal   NodeHasSufficientMemory  17m                   kubelet, node1  Node node1 status is now: NodeHasSufficientMemory
  Normal   NodeHasNoDiskPressure    17m                   kubelet, node1  Node node1 status is now: NodeHasNoDiskPressure
  Normal   NodeHasSufficientPID     17m                   kubelet, node1  Node node1 status is now: NodeHasSufficientPID
```

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/kubelet/oom/oom_watcher_linux.go [hunk 1]
```diff
@@ -58,7 +60,11 @@ func (ow *realWatcher) Start(ref *v1.ObjectReference) error {
 		for event := range outStream {
 			if event.ContainerName == "/" {
 				klog.V(1).Infof("Got sys oom event: %v", event)
-				ow.recorder.PastEventf(ref, metav1.Time{Time: event.TimeOfDeath}, v1.EventTypeWarning, systemOOMEvent, "System OOM encountered")
+				eventMsg := "System OOM encountered"
+				if event.ProcessName != "" && event.Pid != 0 {
+					eventMsg = fmt.Sprintf("%s, victim process: %s, pid: %d", eventMsg, event.ProcessName, event.Pid)
+				}
+				ow.recorder.PastEventf(ref, metav1.Time{Time: event.TimeOfDeath}, v1.EventTypeWarning, systemOOMEvent, eventMsg)
```
Comment (mattjmcnaughton) [inline pkg/kubelet/oom/oom_watcher_linux.go]: Nit: It seems like this information is not only helpful as an event, but would also be great to include in logs. Do you know if all events are also logged somewhere more permanent?

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm
/ok-to-test

I can't see any downside to this... the only potential issue I foresee would be if the `ProcessName` is particularly long, but I assume `PastEventf` has some form of truncation.

File: pkg/kubelet/oom/oom_watcher_linux.go (hunk 1 above)
Comment (gaorong) [reply pkg/kubelet/oom/oom_watcher_linux.go]: there is a log printing about this:

https://github.com/kubernetes/kubernetes/blob/2d5020e3b650c8df92e8a0a6c49426d801055eb7/pkg/kubelet/oom/oom_watcher_linux.go#L62

**Comment by mattjmcnaughton** [comment]
On Wed, May 22, 2019 at 08:26:14PM +0000, Kubernetes Prow Robot wrote:
> @gaorong: The following tests **failed**, say `/retest` to rerun them all:
> 
> Test name | Commit | Details | Rerun command
> --- | --- | --- | ---
> pull-kubernetes-e2e-gce | 2d5020e3b650c8df92e8a0a6c49426d801055eb7 | [link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/78174/pull-kubernetes-e2e-gce/1131277899966451715/) | `/test pull-kubernetes-e2e-gce`
> pull-kubernetes-dependencies | 2d5020e3b650c8df92e8a0a6c49426d801055eb7 | [link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/78174/pull-kubernetes-dependencies/1131277899966451713) | `/test pull-kubernetes-dependencies`
> pull-kubernetes-e2e-gce-100-performance | 2d5020e3b650c8df92e8a0a6c49426d801055eb7 | [link](https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/78174/pull-kubernetes-e2e-gce-100-performance/1131277900062920704/) | `/test pull-kubernetes-e2e-gce-100-performance`
> 
> [Full PR test history](https://prow.k8s.io/pr-history?org=kubernetes&repo=kubernetes&pr=78174). [Your PR dashboard](https://gubernator.k8s.io/pr/gaorong). Please help us cut down on flakes by [linking to](https://git.k8s.io/community/contributors/devel/sig-testing/flaky-tests.md#filing-issues-for-flaky-tests) an [open issue](https://github.com/kubernetes/kubernetes/issues?q=is:issue+is:open) when you hit one in your PR.
> 
> <details>
> 
> Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
> </details>
> 
> 
> -- 
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kubernetes/kubernetes/pull/78174#issuecomment-494958015

/retest

**Comment by gaorong** [comment]
/assign @yujuhong   

please take a look if you have some time

**Comment by dims** [comment]
/milestone v1.15

Dear @kubernetes/sig-node-pr-reviews looks like a good thing to have for better visibility into OOM(s). can you please consider this for 1.15?

**Comment by xmudrii** [comment]
@gaorong @yujuhong @dims Are we still planning this PR for 1.15? Is this a release-blocking PR?

**Comment by gaorong** [comment]
@dashpole could you please take a look at this simple change? as you are in charge of cadvisor related stuff.

**Comment by dashpole** [comment]
This seems reasonable to me.
/lgtm

**Comment by gaorong** [comment]
could someone help me to find out why the `approve` command is not working? I have no idea.

**Comment by ttousai** [comment]
Hello! I'm a bug triage shadow for the 1.16 release cycle and considering this issue is tagged for 1.16, but not updated for a long time, I'd like to check its status. The code freeze is starting on August 29th (about 3 days from now), which means that there should be a PR ready (and merged) by then.

Do we still target this issue to be fixed for 1.16?

**Review by feiskyer — COMMENTED** [review]
super +1 for the event enrichment. @yujuhong could you help to approve again?

**Pull request merged**